doc/whats_new/v1.4.rst

```diff
@@ -151,6 +152,12 @@ Changelog
   `kdtree` and `balltree` values will be removed in 1.6.
   :pr:`26744` by :user:`Shreesha Kumar Bhat <Shreesha3112>`.
 
+:mod:`sklearn.datasets`
+.......................
+
+- |Fix| :func:`datasets.make_sparse_spd_matrix` now returns sparse matrix instead of
+  dense array. :pr:`27438` by :user:`Yao Xiao <Charlie-XIAO>`
```

Comment:
For backward compatibility reasons, we should not change the default behavior because this is a public function and this cannot really be considered as a bug.

So we should document this as an enhancement, add a keyword argument such as `sparse_container_type=None` to call `.toarray()` and return a numpy array by default.

Then in our test we can use the keywoard argument to return the data for the user specified sparse format.